pkg/controller/infinispan/resources/secret/secret_controller.go

```diff
@@ -75,25 +75,42 @@ func Add(mgr manager.Manager) error {
 }
 
 func (s *secretResource) Process() (reconcile.Result, error) {
-	if !s.infinispan.IsGeneratedSecret() {
-		return reconcile.Result{}, nil
+	operatorNs, err := kube.GetOperatorNamespace()
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("Unable to get operator namespace: %w", err)
 	}
-	if err := s.client.Get(context.TODO(), types.NamespacedName{Namespace: s.infinispan.Namespace, Name: s.infinispan.GetSecretName()}, &corev1.Secret{}); err == nil || !errors.IsNotFound(err) {
+
+	secretExists, err := s.secretExists(consts.AdminSecretName, operatorNs)
```

Comment:
I'm not sure that this should work in multi-namespace deployment. Secrets can't be shared between namespaces
https://kubernetes.io/docs/concepts/configuration/secret/#details
Each Infinispan CR should has it's own admin secret

Reply:
The secret is contained within the operator namespace and only the operator accesses it, regardless of where the Infinispan CR is created.

Reply:
You are mounting this Secret into the Pod Volume. This is not possible for multi-namespace installation
```
  volumes:
    - name: data-volume
      persistentVolumeClaim:
        claimName: data-volume-example-infinispan-0
    - name: config-volume
      configMap:
        name: example-infinispan-configuration
        defaultMode: 420
    - name: admin-identities-volume
      secret:
        secretName: infinispan-operator-credentials
        defaultMode: 420
```

Reply:
Great point @dmvolod! I hadn't thought it in that way. So I guess the options are:

1. Have a Secret in the operator namespace with the generated credentials, and then create a copy of this for each Infinispan deployment. The advantages of this is that the same credentials can be used for all clusters and only need to be updated in one place. Updates to the operator secret would have to be reflected across all per cluster secrets.

2. Generate admin secrets per cluster. Simpler, but would require users to create the admin secret per cluster if they wanted to provide custom credentials.

Reply:
Oh, sure, @ryanemerson I understand your concern.
Please have a look at pros and cons of each solution.

> * Have a Secret in the operator namespace with the generated credentials, and then create a copy of this for each Infinispan deployment. The advantages of this is that the same credentials can be used for all clusters and only need to be updated in one place. Updates to the operator secret would have to be reflected across all per cluster secrets.

pros

- single Secret, easy to manage

cons

- complex code
- need to have separate operator Secret controller
- better to store master Secret in-memory if someone drop it (for better restoring)
- able to create an option to have admin secret per Infinispan CR (some customer want to have this option due to security reasons)

> * Generate admin secrets per cluster. Simpler, but would require users to create the admin secret per cluster if they wanted to provide custom credentials.

pros

- easy to implement

cons

- need to maintain each admin Secret (if needed) individually

Reply:
I think for the initial implementation lets keep it simple and generate a secret per cluster.

Reply:
I've added a commit with these changes.

Reply:
Good, thanks.